src/api/mapping.ts

```diff
@@ -11,6 +11,7 @@ const exec = util.promisify(cp.exec)
 
 mappingRouter.post('/', (req, res) => {
   const domainKeys = getMappings()
+  if(!req.body.ip) req.body.ip = '127.0.0.1'
```

Comment:
`req.body` is part of the request, and request does not have `req.body.ip`, so you should keep it that way. 

Instead, below you should just do `ip: req.body.ip || '127.0.0.1'`